### PR TITLE
Tests for audit of Temporal user code calls, part 1

### PR DESCRIPTION
**Pull request by ptomato**
tc39/proposal-temporal#2519 is a very large pull request and I'm planning to land it in parts. This PR contains tests for the first part. Each commit here corresponds to the same-named commit in the proposal-temporal normative PR.

### CODE REVIEW

File: test/built-ins/Temporal/PlainYearMonth/from/observable-get-overflow-argument-string.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+TemporalHelpers.assertPlainYearMonth(result, 2021, 5, "M05");
+
+actual.splice(0);  // empty it for the next check
+const failureExpected = [
+  "ownKeys options",
+  "getOwnPropertyDescriptor options.overflow",
+  "get options.overflow",
+];
+
+assert.throws(TypeError, () => Temporal.PlainYearMonth.from(7, options));
```
Comment (Ms2ger) [inline test/built-ins/Temporal/PlainYearMonth/from/observable-get-overflow-argument-string.js:33]: For all of these, perhaps `-argument-primitive.js` is a better name

File: test/built-ins/Temporal/PlainYearMonth/prototype/with/calendar-arguments.js [hunk 2]
```diff
@@ -12,15 +12,18 @@ includes: [temporalHelpers.js]
...
 class CustomCalendar extends Temporal.Calendar {
   constructor() {
     super("iso8601");
   }
   yearMonthFromFields(...args) {
     assert.sameValue(args.length, 2, "args.length");
     assert.sameValue(typeof args[0], "object", "args[0]");
-    assert.sameValue(args[1], options, "args[1]");
+    assert.notSameValue(args[1], options, "args[1] is a copy of options");
+    assert.sameValue(args[1].extra, "property", "All properties are copied");
```
Comment (Ms2ger) [inline test/built-ins/Temporal/PlainYearMonth/prototype/with/calendar-arguments.js:26]: Consider whether you want to verify the prototype for these

Comment (ptomato) [reply test/built-ins/Temporal/PlainYearMonth/prototype/with/calendar-arguments.js:26]: Added, and found that ZonedDateTime didn't have a test like this one yet, so added that as well.

**Comment by ptomato** [comment]
Thanks for the review!

**Pull request merged**